packages/vue-app/template/utils.js

```diff
@@ -146,7 +146,19 @@ export function resolveRouteComponents (route, fn) {
     flatMapComponents(route, async (Component, instance, match, key) => {
       // If component is a function, resolve it
       if (typeof Component === 'function' && !Component.options) {
-        Component = await Component()
+        try {
+          Component = await Component()
+        } catch (error) {
+          // Handle webpack chunk loading errors
+          // This may be due to a new deployment or a network problem
+          if (error && error.name === 'ChunkLoadError' && window.location.hash !== '#retry') {
+            // mark the page not to reload infinitely
+            window.location.hash = '#retry'
```

Comment:
Modification of URL, might interference with application logic. We may use [navigation type](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigation/type) guard. (safari workaround needed)

Reply:
There's no clean way to solve this, can't think of anything at least. Navigation type seems unreliable. We could rename this retry to #__nuxt_retry or something. Or try to use sessionStorage for this?

We could not have this check at all but there's the unlikely possibility of a currupt build where a critical file is actually missing from the file system. Infinite reloads will not help when debugging that.

Reply:
Do you have any reference regarding the unreliability of nav type? But of course we can detect incompatible UA and switch to localStorage (or sessionStorage) method.   Modification of `#` overrides one if set in real application.

Reply:
As you pointed out navigation type is not supported by Safari so I can't rely on it. I'm working on a sessionStorage method.